src/Npgsql/Util/StreamExtensions.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Buffers;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Npgsql.Util
+{
+    // Helpers to read/write Span/Memory<byte> to Stream before netstandard 2.1
+    internal static class StreamExtensions
+    {
+#if NET461 || NETSTANDARD2_0
+        public static int Read(this Stream stream, Span<byte> buffer)
+        {
+            var sharedBuffer = ArrayPool<byte>.Shared.Rent(buffer.Length);
+            try
+            {
+                var numRead = stream.Read(sharedBuffer, 0, buffer.Length);
+                if ((uint)numRead > (uint)buffer.Length)
```

Comment:
This is new to me - can streams read more than requested?

Reply:
They should not...
Removed.